freeipa/src/main/java/com/sequenceiq/freeipa/sync/ProviderChecker.java

```diff
@@ -53,17 +57,23 @@
             });
             checkableInstances.forEach(instanceMetaData -> {
                 if (statuses.stream().noneMatch(s -> s.getCloudInstance().getInstanceId().equals(instanceMetaData.getInstanceId()))) {
-                    if (updateStatus) {
-                        setStatusIfNotTheSame(instanceMetaData, InstanceStatus.DELETED_ON_PROVIDER_SIDE);
-                        instanceMetaDataService.save(instanceMetaData);
+                    if (!flowLogService.isOtherFlowRunning(stack.getId())) {
```

Comment:
it's strange we need this. I think `StackStatusCheckerJob` already checks

Reply:
It can take long seconds to fetch provider side information. Meanwhile, flow can start in the background.

Reply:
in that case it would make sense to throw an exception or similar to stop the whole sync. right now we will continue the sync just log an error.

Reply:
also I don't know if it's a good idea to have this check in a loop. freeipa won't have many instances but I would move these checks out from loops to ease the load on the database

Reply:
you are right:
- I created a new exception (InterruptSyncingException)
- check is moved before loops but after provider query operation

Reply:
thanks for the changes!
please consider catching `InterruptSyncingException` in `executeInternal` method instead of letting it propagate to quartz. There is a try-finally block already

Reply:
good idea!